crates/uv-distribution-types/src/index.rs

```diff
@@ -155,6 +155,21 @@ impl Index {
         self.url.url()
     }
 
+    /// Return the raw [`URL`] of the index's `/files` endpoint, if any.
+    ///
+    /// We only assume there is a `/files` endpoint if the URL ends with `/simple`.
+    pub fn files_url(&self) -> Option<Url> {
+        let segments = self.raw_url().path_segments()?;
+        let last = segments.last()?;
+        if !last.eq_ignore_ascii_case("simple") {
+            return None;
+        }
+
+        let mut url = self.raw_url().clone();
+        url.path_segments_mut().ok()?.pop().push("files");
+        Some(url)
+    }
```

Comment:
Alternatively, we could just push credentials for the URL with `/simple` stripped and that would cover all imaginable cases?